compiler/rustc_hir_analysis/src/collect.rs

```diff
@@ -416,6 +416,15 @@ impl<'tcx> AstConv<'tcx> for ItemCtxt<'tcx> {
         self.tcx().const_error_with_message(ty, span, "bad placeholder constant")
     }
 
+    fn effect_infer(
+        &self,
+        kind: ty::EffectKind,
+        _: Option<&ty::GenericParamDef>,
+        _span: Span,
+    ) -> ty::Effect<'tcx> {
+        self.tcx().mk_effect(ty::EffectValue::Rigid { on: false }, kind)
```

Comment:
should be a delay_span_bug or even a full `bug!` (realized that `ty_infer` and `ct_infer` should also probably be a hard error)

Reply:
the others are actually reachable in case of earlier errors, but we should def look into them and see if we can turn them into bug!